src/Tests/dotnet-watch.Tests/BrowserLaunchTests.cs

```diff
@@ -0,0 +1,87 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.IO;
+using System.Threading.Tasks;
+using Microsoft.NET.TestFramework;
+using Xunit.Abstractions;
+
+namespace Microsoft.DotNet.Watcher.Tools
+{
+    public class BrowserLaunchTests
+    {
+        private const string AppName = "WatchBrowserLaunchApp";
+
+        private readonly TestAssetsManager _testAssetsManager;
+        private readonly ITestOutputHelper _logger;
+
+        public BrowserLaunchTests(ITestOutputHelper logger)
+        {
+            _testAssetsManager = new TestAssetsManager(logger);
+            _logger = logger;
+        }
+
+        [PlatformSpecificFact(Xunit.TestPlatforms.Windows | Xunit.TestPlatforms.OSX)]
+        public async Task LaunchesBrowserOnStart()
+        {
+            var expected = "watch : Launching browser: https://localhost:5001/";
+            var testAsset = _testAssetsManager.CopyTestAsset(AppName)
+                .WithSource()
+                .Path;
+
+            using var app = new WatchableApp(testAsset, _logger);
+
+            app.DotnetWatchArgs.Add("--verbose");
+
+            await app.StartWatcherAsync();
+
+            // Verify we launched the browser.
+            await app.Process.GetOutputLineStartsWithAsync(expected, TimeSpan.FromMinutes(2));
```

Comment:
Getting an exception locally here. It looks like it doesn't repro on the Darwin build in CI so not sure if it is reproable. Might be a heads up as to the flakiness of this test case.

```
  Failed Microsoft.DotNet.Watcher.Tools.BrowserLaunchTests.LaunchesBrowserOnStart [2 m 9 s]
  Error Message:
   System.Threading.Tasks.TaskCanceledException : A task was canceled.
  Stack Trace:
     at Microsoft.DotNet.Watcher.Tools.AwaitableProcess.GetOutputLineAsync(String predicateName, Predicate`1 predicate, CancellationToken cancellationToken) in /Users/captainsafia/github.com/dotnet/sdk/src/Tests/dotnet-watch.Tests/Utilities/AwaitableProcess.cs:line 94
   at Microsoft.DotNet.Watcher.Tools.AwaitableProcess.GetOutputLineStartsWithAsync(String message, TimeSpan timeout) in /Users/captainsafia/github.com/dotnet/sdk/src/Tests/dotnet-watch.Tests/Utilities/AwaitableProcess.cs:line 87
   at Microsoft.DotNet.Watcher.Tools.BrowserLaunchTests.LaunchesBrowserOnStart() in /Users/captainsafia/github.com/dotnet/sdk/src/Tests/dotnet-watch.Tests/BrowserLaunchTests.cs:line 40
--- End of stack trace from previous location ---
```